signer/tests/integration/complete_deposit.rs

```diff
@@ -81,119 +78,54 @@ fn make_complete_deposit(
-}
-
-/// For this test we check that the `CompleteDepositV1::validate` function
-/// returns okay when everything matches the way that it is supposed to.
 #[cfg_attr(not(feature = "integration-tests"), ignore)]
 #[tokio::test]
 async fn complete_deposit_validation_happy_path() {
     // Normal: this generates the blockchain as well as deposit request
-    // transactions in each bitcoin block.
+    // transactions and a transaction sweeping in the deposited funds.
+    // This is just setup and should be essentially the same between tests.
     let db_num = DATABASE_NUM.fetch_add(1, Ordering::SeqCst);
     let db = testing::storage::new_test_database(db_num, true).await;
     let mut rng = rand::rngs::StdRng::seed_from_u64(51);
-    let signatures_required = 2;
-
-    let signer_set = deposit_setup(&mut rng, &db).await;
-    // Get the chain tip.
-    let chain_tip = get_bitcoin_canonical_chain_tip_block(&db).await;
-    // Normal: Get an existing deposit request on the canonical bitcoin
-    // blockchain.
-    let deposit_req =
-        get_pending_accepted_deposit_requests(&db, &chain_tip, signatures_required).await;
-
-    // Normal: we generate a transaction that sweeps in the deposit.
-    let sweep_config = SweepTxConfig {
-        aggregate_key: PublicKey::combine_keys(&signer_set).unwrap(),
-        amounts: 3000..1_000_000_000,
-        inputs: vec![deposit_req.outpoint()],
-        outputs: Vec::new(),
-    };
-    let mut sweep_tx: model::Transaction = sweep_config.fake_with_rng(&mut rng);
-    // Normal: make sure the sweep transaction is on the canonical bitcoin
-    // blockchain and is in our database.
-    sweep_tx.block_hash = chain_tip.block_hash.into_bytes();
-    // Normal: make sure that we have a record of the sweep transaction in
-    // our database.
-    let bitcoin_tx_ref = BitcoinTxRef {
-        txid: sweep_tx.txid.into(),
-        block_hash: sweep_tx.block_hash.into(),
-    };
-    db.write_transaction(&sweep_tx).await.unwrap();
-    db.write_bitcoin_transaction(&bitcoin_tx_ref).await.unwrap();
-
-    // Generate the transaction and corresponding request context.
-    let (complete_deposit_tx, req_ctx) = make_complete_deposit(&deposit_req, &sweep_tx, &chain_tip);
-
-    // This should not return an Err.
+    let (rpc, faucet) = regtest::initialize_blockchain();
+    let setup = TestSweepSetup::new_setup(&rpc, &faucet, 1_000_000, &mut rng);
+
+    // Normal: the signer follows the bitcoin blockchain and event observer
+    // should be getting new block events from bitcoin-core. We haven't
+    // hooked up our block observer so we need to manually update the
+    // database with new bitcoin block headers.
+    backfill_bitcoin_blocks(&db, rpc, &setup.sweep_block_hash).await;
+
+    // Normal: we take the deposit transaction as is from the test setup
+    // and store it in the database. This is necessary for when we fetch
+    // outstanding unfulfilled deposit requests.
+    setup.store_deposit_tx(&db).await;
```

Comment:
At the moment, I do not think that this step is actually necessary since we get the deposit transaction from bitcoin-core. I discovered this along the way and just decided to keep it in, but I don't mind removing it.

Reply:
I think this hangs a little together with @matteojug's Q regarding using multiple data-sources.